pkg/kubelet/cm/dra/manager.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/dynamic-resource-allocation/resourceclaim"
+	"k8s.io/klog/v2"
+	dra "k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+// ManagerImpl is the structure in charge of managing DRA resource Plugins.
+type ManagerImpl struct {
+	sync.Mutex
+
+	// resources contains resources referenced by pod containers
+	resources *claimedResources
```

Comment:
should this variable be called `preparedResources` to make it clear that it holds a reference to resources that have already been prepared.

Reply:
After reviewing `resources.go` I think this variable should be declared as:
```
cache `claimInfoCache`
```

Reply:
Resolved